__init__.py

```diff
@@ -10,19 +10,196 @@
+            return data
+
+        # extract the best keyword with some regexes or fallback to RAKE
+        kw = self.extract_keyword(query, lang)
+        return self.get_data(kw, context)
+
+    # officially exported Solver methods
+    def get_data(self, query, context):
+        """
+        query assured to be in self.default_lang
+        return a dict response
+        """
+        # duck duck go api request
+        try:
+            data = requests.get("https://api.duckduckgo.com",
+                                params={"format": "json",
+                                        "q": query}).json()
+        except:
+            return {}
+        return data
+
+    def get_image(self, query, context=None):
+        """
+        query assured to be in self.default_lang
+        return path/url to a single image to acompany spoken_answer
+        """
+        data = self.extract_and_search(query, context)
+        image = data.get("Image") or \
+                "https://github.com/JarbasSkills/skill-ddg/raw/master/ui/logo.png"
+        if image.startswith("/"):
+            image = "https://duckduckgo.com" + image
+        return image
+
+    def get_spoken_answer(self, query, context=None):
+        """
+        query assured to be in self.default_lang
+        return a single sentence text response
+        """
+        query, context, lang = self._tx_query(query, context)
+
+        # match an infobox field with some basic regexes
+        # (primitive intent parsing)
+        selected, key = DDGIntents.match(query, lang)
+
+        if key:
+            selected = self.extract_keyword(selected, lang)
+            infobox = self.get_infobox(selected, context)[0] or {}
+            answer = infobox.get(key)
+            if answer:
+                return answer
+
+        # return summary
+        data = self.extract_and_search(query, context)
+        return data.get("AbstractText")
+
+    def get_expanded_answer(self, query, context=None):
+        """
+        query assured to be in self.default_lang
+        return a list of ordered steps to expand the answer, eg, "tell me more"
+
+        {
+            "title": "optional",
+            "summary": "speak this",
+            "img": "optional/path/or/url
+        }
+        :return:
+        """
+        data = self.get_data(query, context)
+        img = self.get_image(query, context)
+        steps = [{
+            "title": query,
+            "summary": s,
+            "img": img
+        } for s in self.sentence_split(data.get("AbstractText", ""), -1) if s]
+
+        infobox, _ = self.get_infobox(query)
+        steps += [{"title": k,
+                   "summary": k + " - " + str(v),
+                   "img": img} for k, v in infobox.items()
+                  if not k.endswith(" id") and  # itunes id
+                  not k.endswith(" profile") and  # twitter profile
+                  k != "instance of"]  # spammy and sounds bad when spokem
+        return steps
```

Comment:
Is there a way to make the search excluding iTunes and Twitter, similar to how Google allows it, rather than removing it here?